docs/source/image_dataset_script.mdx

```diff
@@ -0,0 +1,253 @@
+* Upload the dataset to the Hub.
+
+The best way to learn is to open up an existing image dataset loading script, like [Food-101](https://huggingface.co/datasets/food101/blob/main/food101.py), and follow along!
+
+<Tip>
+
+To help you get started, we created a loading script [template](https://github.com/huggingface/datasets/blob/main/templates/new_dataset_script.py) you can copy and use as a starting point!
+
+</Tip>
+
+## Create a dataset builder class
+
+[`GeneratorBasedBuilder`] is the base class for datasets generated from a dictionary generator. Within this class, there are three methods to help create your dataset:
+
+* `info` stores information about your dataset like its description, license, and features.
+* `split_generators` downloads the dataset and defines its splits.
+* `generate_examples` generates the images and labels for each split.
+
+Start by creating your dataset class as a subclass of [`GeneratorBasedBuilder`] and add the three methods. Don't worry about filling in each of these methods yet, you'll develop those over the next few sections:
+
+```py
+>>> class Food101(datasets.GeneratorBasedBuilder):
+...     """Food-101 Images dataset"""
+...
+...     def _info(self):
+...
+...     def _split_generators(self, dl_manager):
+...
+...     def _generate_examples(self, images, metadata_path):
+```
+
+### Multiple configurations
+
+In some cases, a dataset may have more than one configuration. For example, if you check out the [Imagenette dataset](https://huggingface.co/datasets/frgfm/imagenette), you'll notice there are three subsets. 
+
+To create different configurations, use the [`BuilderConfig`] class to create a subclass for your dataset. Provide the links to download the images and labels in `data_url` and `metadata_urls`:
+
+```py
+>>> class Food101Config(datasets.BuilderConfig):
+...     """Builder Config for Food-101"""
+... 
+...     def __init__(self, data_url, metadata_urls, **kwargs):
+...         """BuilderConfig for Food-101.
+...         Args:
+...           data_url: `string`, url to download the zip file from.
+...           metadata_urls: dictionary with keys 'train' and 'validation' containing the archive metadata URLs
+...           **kwargs: keyword arguments forwarded to super.
+...         """
+...         super(Food101Config, self).__init__(version=datasets.Version("1.0.0"), **kwargs)
+...         self.data_url = data_url
+...         self.metadata_urls = metadata_urls
+```
+
+Now you can define your subsets at the top of [`GeneratorBasedBuilder`]. Imagine you want to create two subsets in the Food-101 dataset based on whether it is a breakfast or dinner food.
+
+1. Define your subsets with `Food101Config` in a list in `BUILDER_CONFIGS`.
+2. For each configuration, provide a name, description, and where to download the images and labels from.
+
+```py
+>>> class Food101(datasets.GeneratorBasedBuilder):
+...     """Food-101 Images dataset"""
+... 
+...     BUILDER_CONFIGS = [
+...         Food101Config(
```

Comment:
It's maybe better to not use the `>>>`/`...` syntax here, to make it easy for people to copy paste into a .py file ?

Reply:
This syntax should be ignored if you use the copy-to-clipboard button.

Reply:
The syntax is also ignored if you try to copy and paste it normally. But I removed it for code snippets that are being copied into a `py` file since I don't think it needs to be doctested :)